mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,270 @@
+      if (BY_VALUE_MAP.containsKey(value)) {
+        return BY_VALUE_MAP.get(value);
+      } else {
+        return RequestContentType.Invalid;
+      }
+    }
+  }
+
+  private final Optional<Predictor> predictor;
+  private final Optional<Integer> portNumber;
+  private Optional<Service> activeService = Optional.empty();
+
+  /**
+   * Constructs a {@link ScoringServer} to serve the specified {@link Predictor} on the local host
+   * at the default port: {@link ScoringServer#DEFAULT_PORT}
+   */
+  public ScoringServer(Predictor predictor) {
+    this(Optional.of(predictor), Optional.empty());
+  }
+
+  /**
+   * Constructs a {@link ScoringServer} to serve the specified {@link Predictor} on the local host
+   * at the specified port
+   */
+  public ScoringServer(Predictor predictor, int portNumber) {
+    this(Optional.of(predictor), Optional.of(portNumber));
+  }
+
+  /**
+   * Loads the MLFlow model at the specified path as a {@link Predictor} and serves it on the local
+   * host at the default port: {@link ScoringServer#DEFAULT_PORT}
+   *
+   * @param modelPath The path to the MLFlow model to serve
+   */
+  public ScoringServer(String modelPath) throws IOException, PredictorLoadingException {
+    this(modelPath, Optional.empty(), true);
+  }
+
+  /**
+   * Loads the MLFlow model at the specified path as a {@link Predictor} and serves it on the local
+   * host at the specified port
+   *
+   * @param modelPath The path to the MLFlow model to serve
+   */
+  public ScoringServer(String modelPath, int portNumber)
+      throws IOException, PredictorLoadingException {
+    this(modelPath, Optional.of(portNumber), true);
+  }
+
+  /**
+   * Loads the MLFlow model at the specified path as a {@link Predictor} and serves it on the local
+   * host at the specified port
+   *
+   * @param modelPath The path to the MLFlow model to serve
+   * @param failOnUnsuccessfulModelLoad If `true`, an exception will be thrown if the specified
+   *     model cannot be loaded. If `false`, the server will still be constructed, and it will
+   *     respond to pings and invocations with an error message indicating that the model could not
+   *     be loaded
+   */
+  protected ScoringServer(String modelPath, int portNumber, boolean failOnUnsuccessfulModelLoad)
+      throws IOException, PredictorLoadingException {
+    this(modelPath, Optional.of(portNumber), failOnUnsuccessfulModelLoad);
+  }
+
+  private ScoringServer(
+      String modelPath, Optional<Integer> portNumber, boolean failOnUnsuccessfulModelLoad)
+      throws IOException, PredictorLoadingException {
+    this(loadPredictorFromPath(modelPath, failOnUnsuccessfulModelLoad), portNumber);
+  }
+
+  private ScoringServer(Optional<Predictor> predictor, Optional<Integer> portNumber) {
+    this.predictor = predictor;
+    this.portNumber = portNumber;
+  }
+
+  private static Optional<Predictor> loadPredictorFromPath(
+      String modelPath, boolean failOnUnsuccessfulModelLoad)
```

Comment:
Change to return Predictor and assume failOnUnsuccessfulModelLoad = true.

Reply:
Done!